code/PostgreSQLQuery.php

```diff
@@ -47,6 +64,24 @@ public function numRecords()
 
     public function nextRecord()
     {
-        return pg_fetch_assoc($this->handle);
+        $row = pg_fetch_array($this->handle, null, PGSQL_NUM);
+
+        // Correct non-string types
+        if ($row) {
+            $record = [];
+
+            foreach ($row as $i => $v) {
+                $k = $this->columnNames[$i];
+                $record[$k] = $v;
+                $type = pg_field_type($this->handle, $i);
+                if (isset(self::$typeMapping[$type])) {
```

Comment:
Is `$typeMapping` designed to be a config prop (because it's a private static)? If not, maybe we could use protected to distinguish the difference? If it is, we should use `self::config()->get('typeMapping')`